hardware_interface/src/lexical_casts.cpp

```diff
@@ -16,6 +16,8 @@
 
 namespace hardware_interface
 {
+
+#if __cplusplus > 201402L
```

Comment:
[`std::from_chars`](https://en.cppreference.com/w/cpp/utility/from_chars) is available since C++17, but doesn't this logic tell us to use the `stod` implementation with `std::from_chars` for anything below C++14?

I think the logic should read like this instead:

```cpp
#if __cplusplus < 201703L
double stod(const std::string & s)
{
    // Impl with std::istringstream
}
#else
double stod(const std::string & s)
{
    // Impl with std::from_chars
}
#endif
```

Reply:
just a small note: [std::from_chars](https://en.cppreference.com/w/cpp/utility/from_chars) was not implemented in every C++17 compiler because it was added very late to the standard. We can assure that only for versions newer than C++17

Reply:
for better assurance, made the cpp version check for newer than C++ 17.